pulp/rpm.py

```diff
@@ -0,0 +1,66 @@
+import os
+from functools import wraps
+
+from flask import make_response, request
+from werkzeug.utils import secure_filename
+
+import features
+
+from app import app
+from auth.decorators import process_basic_auth_no_pass
+from auth.registry_jwt_auth import process_registry_jwt_auth
+from endpoints.v2 import require_repo_write
+from endpoints.decorators import (
+    anon_protect,
+    check_pushes_disabled,
+    check_readonly,
+    disallow_for_account_recovery_mode,
+    parse_repository_name,
+    route_show_if,
+)
+from pulp import pulp_bp, pulp_client
+
+# UPLOAD_FOLDER = '/quay-registry'
+# app.config['UPLOAD_FOLDER'] = UPLOAD_FOLDER
+
+
+BASE_PULP_PLUGIN_RPM_ROUTE = "/<repopath:repository>/_pulp/rpm"
+
+
+@pulp_bp.route(BASE_PULP_PLUGIN_RPM_ROUTE, methods=["GET"])
+@parse_repository_name()
+@anon_protect
+def pulp_plugin_support(namespace_name, repo_name):
+    raise NotImplementedError()
+
+
+@pulp_bp.route(BASE_PULP_PLUGIN_RPM_ROUTE, methods=["POST"])
+@route_show_if(features.PULP_PLUGIN)
+@disallow_for_account_recovery_mode
+@parse_repository_name()
+@process_basic_auth_no_pass
+@require_repo_write(allow_for_superuser=False, disallow_for_restricted_users=True)
+@anon_protect
+@check_readonly
+@check_pushes_disabled
+def pulp_publish_rpm(namespace_name, repo_name):
+    ref_name = "/".join([namespace_name, repo_name])
+
+    pulp_repo = pulp_client.get_repository(ref_name)
+    if pulp_repo.json().get("count") == 0:
+        assert pulp_client.create_repository(ref_name).status_code // 100 == 2
+
+    repository_prn = pulp_repo.json()["results"][0]["prn"]
```

Comment:
If the repository count is 0 (`pulp_repo.json().get("count") == 0`) won't this result in an index error? Do we need another variable for the created repo on line 52 and reference that in the scenario we're creating it on demand?

Reply:
I'm not sure why it would cause an index error. If count is already 0, then we call the repo creation function. If it fails, the assertion will fail as well and an exception will be raised and execution will stop. If everything passes, then `repository_prn` should contain something after execution, `["results"][0]["prn"]` should not be `Null` but should have some value.

Reply:
won't `pulp_repo.json()` will still contain the results from the first call when the repo didn't exist? So `len(pulp_repo.json()["results"]) == 0`

Reply:
Oh, now I get what you're saying. Yeah, we're not saving the result from the `create_repo` function.